README.md

```diff
@@ -1 +1,167 @@
+        if (err) {
+          console.log('Error identifying file size: ' + err)
+        } else {
+          console.log(filename + ' : ' + values)
+          aspect = (values.width / values.height)
+          widths.forEach(function (width, widthIndex) {
+            height = Math.round(width / aspect)
+            console.log('resizing ' + filename + 'to ' + height + 'x' + height)
+            this.resize(width, height).write(dest + 'w' + width + '_' + filename, function(err) {
+              if (err) console.log('Error writing file: ' + err)
+            })
+          }.bind(this))
+        }
+      })
+    })
+  }
+}
+```
+
+You might end up with something called the nested approach, pyramid of doom or callback hell. This code is very hard to read and maintain. People who have to work on this code after you might start to hate you!
+
+The `readdir` example is from <http://callbackhell.com> (yes, there is even a website for Callback Hell!).
+
+This lesson will cover the ways to combat callback hell.
+
+## Objectives
+
+1. Explain that a callback hell script is a complication of the callback async pattern
+1. Identify a callback hell script
+1. Implement possible ways to combat callback hell with named functions
+1. Implement possible ways to combat callback hell with modules
+
+
+## Using A Named Function to Mitigate Callback Hell
+
+The easiest way to reduce the level of indentation and thus reduce callback hell is to use named functions. Named functions are functions with names or variables which values are functions.
+
+```js
+function processResponse1(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+or
+
+```js
+var processResponse1 = function(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+The difference is that the named function will be hoisted, meaning its definition can be anywhere in the file while the variable function's definition must precede the usage of it. Another difference between named functions and variables is that named function will have names in the stack trace (the error message for debugging) so they are better for trouble shooting and debugging..
+
+Now where did the other two request go? They are inside of the `processResponse1`:
+
+```js
+callOne({...}, processResponse1)
+
+function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+
+function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+
+function processResponse3(error, data1) {
+  ...
+}
+```
+
+This code is considerably easier to read, but that's not all we can do to improve it. We can take it a step further.
+
+Note: `callOne`, `callTwo`, etc. are just pseudo code for making HTTP requests. In real-life example you would substitute them for real functions using libraries like `http`, `request` or `superagent`. For now it's not really important what those functions do as long as they are asynchronous and use callbacks.
```

Comment:
Yeah, let's make them real. Something like commenting on a file in Google Drive? https://developers.google.com/apis-explorer/#p/drive/v3/drive.comments.create